### ISSUE TITLE
Our gif proxying sucks

### ISSUE THREAD

**Issue by huumn** [stackernews/stacker.news#444]
The majority of gifs people share timeout while being processed by the proxy. The UX is abysmal:

- nothing shows for 10+ seconds
- the viewer scrolls or navigates elsewhere
- if they sit there we timeout and then it loads

I've given our containers more memory and cpu in the hopes that that improves things but it'd be nice to have something that works reliabely.

**Comment by SatsAllDay** [comment]
Is it just because of the file size? I agree some kind of loading indicator would be ideal

**Comment by huumn** [comment]
It's the nature of gifs. It's like processing 100 images at once.

It seems like bumping the CPU/memory of the proxy helped some but it'd be nice to handle failure cases more gracefully

**Comment by ekzyis** [comment]
> if they sit there we timeout and then it loads

This means they have enabled automatically to load external images, right?

We could maybe just load both requests in parallel or just drop the image proxy all together for users who would load external images anyway ("click to load external images" disabled in settings).

On another note, the current "click to load image" is also not good UX. It gives stackers no information why they have to click at all and if they know it's because of external images, we give them no information for an informed decision if they want to load that external image or not. We could at least show the external domain.

**Comment by ekzyis** [comment]
GIFs are also cut off. Only first 10 frames are shown here: https://stacker.news/items/249778

**Comment by ekzyis** [comment]
I think we could improve performance by using lazy loading (so the image proxy does not get hit with 60+ unnecessary requests at the same time whenever someone loads Meme Monday) and using responsive images with `srcset` for proper optimization and scaling:

![2023-09-11-223706_1920x1080_scrot](https://github.com/stackernews/stacker.news/assets/27162016/b1f10b1b-e363-4b46-9b25-0bce378c22a3)

-- https://imagekit.io/website-analyzer?taskId=1694474566205&device=desktop
